src/serac/physics/materials/thermal_material.hpp

```diff
@@ -65,6 +65,62 @@ struct LinearIsotropicConductor {
   double conductivity_;
 };
 
+/// Nonlinear isotropic thermal conduction material model
+struct IsotropicConductorWithLinearConductivityVsTemperature {
+  /**
+   * @brief Construct a Isotropic Conductor with Conductivity linear with Temparture object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material at temp = 0 (power / (length * temp))
+   * @param d_conductivity_d_temp Slope for the thermal conductivity as a function of temperature
+   */
+  IsotropicConductorWithLinearConductivityVsTemperature(double density = 1.0, double specific_heat_capacity = 1.0,
+                                                        double conductivity                 = 1.0,
+                                                        double d_conductivity_d_temperature = 0.0)
+      : density_(density),
+        specific_heat_capacity_(specific_heat_capacity),
+        conductivity_(conductivity),
+        d_conductivity_d_temperature_(d_conductivity_d_temperature)
+  {
+    SLIC_ERROR_ROOT_IF(conductivity_ < 0.0,
+                       "Conductivity must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Material response call for a linear isotropic material
+   *
+   * @tparam T1 Spatial position type
+   * @tparam T2 Temperature type
+   * @tparam T3 Temperature gradient type
+   * @param[in] temperature_gradient Temperature gradient
+   * @return The calculated material response (tuple of volumetric heat capacity and thermal flux) for a linear
+   * isotropic material
+   */
+  template <typename T1, typename T2, typename T3>
+  SERAC_HOST_DEVICE auto operator()(const T1& /* x */, const T2& temperature, const T3& temperature_gradient) const
+  {
+    auto currentConductivity = conductivity_ + d_conductivity_d_temperature_ * temperature;
```

Comment:
I realize the constitutive model is not the point of this PR, but should we have a check here to make sure the conductivity is positive? What is our convention for runtime consistency checks?

Reply:
I changed this to a runtime check, for better or worse.